packages/lib/src/vis/heatmap/HeatmapMesh.tsx

```diff
@@ -1,159 +1,95 @@
 import type { Domain } from '@h5web/shared';
-import { ScaleType, assertDefined, isScaleType } from '@h5web/shared';
+import { ScaleType, isScaleType } from '@h5web/shared';
 import { rgb } from 'd3-color';
 import type { NdArray } from 'ndarray';
 import { memo, useMemo } from 'react';
-import type { TextureDataType, TextureFilter } from 'three';
+import type { TextureFilter } from 'three';
 import { DataTexture, RGBFormat, UnsignedByteType } from 'three';
 
 import { useAxisSystemContext } from '../..';
 import type { VisScaleType } from '../models';
 import VisMesh from '../shared/VisMesh';
-import type { ColorMap, TextureTypedArray, ScaleShader } from './models';
-import { getDataTexture, getInterpolator } from './utils';
+import { DEFAULT_DOMAIN } from '../utils';
+import type { ColorMap } from './models';
+import { getDataTexture, getInterpolator, scaleDomain } from './utils';
 
 interface Props {
-  values: NdArray<TextureTypedArray>;
+  values: NdArray<Float32Array | Uint16Array>; // if `Uint16Array`, it must contain half floats
   domain: Domain;
   scaleType: VisScaleType;
   colorMap: ColorMap;
   invertColorMap?: boolean;
-  textureType?: TextureDataType; // override default texture type (determined from `values.dtype`)
   magFilter?: TextureFilter;
-  alphaValues?: NdArray<TextureTypedArray>;
+  alphaValues?: NdArray<Float32Array>;
```

Comment:
Should be allow `UInt16Array` for `alphaValues` as well ?

Reply:
I guess current main (only?) use is for complex numbers, right?
If so, it is not necessary, but would make the API consistent.